packages/e2e-tests/specs/editor/blocks/navigation.test.js

```diff
@@ -564,6 +573,112 @@ describe( 'Navigation', () => {
 		expect( quickInserter ).toBeTruthy();
 	} );
 
+	it( 'supports navigation blocks that have inner blocks within their markup and converts them to wp_navigation posts', async () => {
+		// Insert 'old-school' inner blocks via the code editor.
+		await createNewPost();
+		await clickOnMoreMenuItem( 'Code editor' );
+		const codeEditorInput = await page.waitForSelector(
+			'.editor-post-text-editor'
+		);
+		await codeEditorInput.click();
+		const markup =
+			'<!-- wp:navigation --><!-- wp:page-list /--><!-- /wp:navigation -->';
+		await page.keyboard.type( markup );
+		await clickButton( 'Exit code editor' );
+		const navBlock = await page.waitForSelector(
+			'nav[aria-label="Block: Navigation"]'
+		);
+		// Select the block to convert to a wp_navigation and publish.
+		// The select menu button shows up when saving is complete.
+		await navBlock.click();
+		await page.waitForSelector( 'button[aria-label="Select Menu"]' );
+		await publishPost();
+
+		// Check that the wp_navigation post has the page list block.
+		expect( await getNavigationMenuRawContent() ).toMatchSnapshot();
+	} );
+
+	describe( 'Creating and restarting', () => {
```

Comment:
This is the end-to-end test added in #36880. It was skipped, but I've re-enabled it here so that it asserts that the changes from that PR still work.

Reply:
Noted these don't provide full coverage for the issue uncovered in https://github.com/WordPress/gutenberg/pull/37358#pullrequestreview-832922134. Hopefully simple to add that coverage though?